File: csharp/acronym/Acronym.cs
using System;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        string acro = "";
        string[] words = phrase.Replace("_", "").Replace("-", " ").Split(" ");
        foreach (var word in words)
        {
            if (word != "")
            {
                acro += word[0].ToString().ToUpper();
            }
        }
        return acro;
    }
}

File: java/acronym/src/main/java/Acronym.java
class Acronym {

    private final String abbreviate;

    Acronym(String phrase) {
        StringBuilder abbr = new StringBuilder();

        final var words = phrase.replace("_", "").replace('-', ' ').split(" ");

        for (var word : words) {
            if (!word.isEmpty()) {
                abbr.append(word.substring(0, 1).toUpperCase());
            }
        }

        abbreviate = abbr.toString();
    }

    String get() {
        return abbreviate;
    }

}
